torch/distributed/_spmd/api.py

```diff
@@ -156,7 +156,7 @@ def _rematerialize_optimizer(
         yield
     finally:
         param_group["params"] = orig_params
-        opt.state.update(orig_states)
+        opt.state = orig_states
```

Comment:
this is assuming no state changes during `.step()` call? I think this is OK for now, as we cannot capture state creation ops correctly anyway in the current impl.

Reply:
Yep it assumes it is after lazy state materialization, I think it's a legit change as we should not capture the state creation ops in the graph, they should be feed in as inputs instead.